code/game/machinery/doors/firedoor.dm

```diff
@@ -140,4 +176,148 @@
 	icon = 'icons/obj/doors/Doorfire.dmi'
 	glass = 0
 
+/obj/item/weapon/firelock_board
+	name = "firelock circuitry"
+	desc = "A circuit board used in construction of firelocks."
+	icon = 'icons/obj/module.dmi'
+	icon_state = "mainboard"
+	w_class = 2
+	materials = list(MAT_METAL = 50, MAT_GLASS = 50)
+
+/obj/structure/firelock_frame
+	name = "firelock frame"
+	desc = "A partially completed firelock."
+	icon = 'icons/obj/doors/Doorfire.dmi'
+	icon_state = "frame1"
+	anchored = 0
+	density = 1
+	var/constructionStep = CONSTRUCTION_NOCIRCUIT
+	var/reinforced = 0
+
+/obj/structure/firelock_frame/examine(mob/user)
+	..()
+	switch(constructionStep)
+		if(CONSTRUCTION_PANEL_OPEN)
+			user << "There is a small metal plate covering the wires."
+		if(CONSTRUCTION_WIRES_EXPOSED)
+			user << "Wires are trailing from the maintenance panel."
+		if(CONSTRUCTION_GUTTED)
+			user << "The circuit board is visible."
+		if(CONSTRUCTION_NOCIRCUIT)
+			user << "There are no electronics in the frame."
+
+/obj/structure/firelock_frame/update_icon()
+	..()
+	icon_state = "frame[constructionStep]"
 
+/obj/structure/firelock_frame/attackby(obj/item/weapon/C, mob/user)
+	switch(constructionStep)
+		if(CONSTRUCTION_PANEL_OPEN)
+			if(istype(C, /obj/item/weapon/crowbar))
+				playsound(get_turf(src), 'sound/items/Crowbar.ogg', 50, 1)
+				user.visible_message("<span class='notice'>[user] starts prying something out from [src]...</span>", \
+									 "<span class='notice'>You begin prying out the wire cover...</span>")
+				if(!do_after(user, 50)) return

```

Comment:
you need a constructionStep check after the do_after to prevent getting the success message multiple times if you spamclick with a crowbar. Also, the do_after needs a target argument!